src/Session.php

```diff
@@ -1016,6 +1024,12 @@ public static function checkValidSessionId()
             );
             if ($result->current()['count'] === 0) {
                 $valid_user = false;
+            } elseif (
+                $result->current()['last_rights_update'] !== null
+                && $result->current()['last_rights_update'] > $_SESSION['glpiactiveprofile']['last_rights_update'] ?? 0
+            ) {
+                Session::reloadCurrentProfile();
+                $_SESSION['glpiactiveprofile']['last_rights_update'] = $result->current()['last_rights_update'];
```

Comment:
You should add a new variable to prevent calling `$result->current()` multiple times.
```suggestion
            $row = $result->current();
            if ($row['count'] === 0) {
                $valid_user = false;
            } elseif (
                $row['last_rights_update'] !== null
                && $row['last_rights_update'] > $_SESSION['glpiactiveprofile']['last_rights_update'] ?? 0
            ) {
                Session::reloadCurrentProfile();
                $_SESSION['glpiactiveprofile']['last_rights_update'] = $row['last_rights_update'];
```